pkg/daemon/daemon.go

```diff
@@ -0,0 +1,97 @@
+package daemon
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"os"
+)
+
+// CreateServer creates a new HTTP server with TLS configured for GPTScript.
+// This function should be used when creating a new server for a daemon tool.
+// The server should then be started with the StartServer function.
+func CreateServer() (*http.Server, error) {
+	tlsConfig, err := getTLSConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get TLS config: %v", err)
+	}
+
+	server := &http.Server{
+		Addr:      fmt.Sprintf("127.0.0.1:%s", os.Getenv("PORT")),
+		TLSConfig: tlsConfig,
+	}
+	return server, nil
+}
+
+// CreateServerWithMux creates a new HTTP server with TLS configured for GPTScript.
+// This function should be used when creating a new server for a daemon tool with a custom ServeMux.
+// The server should then be started with the StartServer function.
+func CreateServerWithMux(mux *http.ServeMux) (*http.Server, error) {
+	tlsConfig, err := getTLSConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get TLS config: %v", err)
+	}
+
+	server := &http.Server{
+		Addr:      fmt.Sprintf("127.0.0.1:%s", os.Getenv("PORT")),
```

Comment:
We probably want to make the `port` a parameter. Otherwise, you could only ever run one daemon at a time.

Reply:
But we do only want to run one daemon at a time, at least one per process, and it is required to use the port that GPTScript chose for it. So I don't think there is a point in making this a parameter. This is just a library function to be called in each daemon tool, not something that will get used anywhere else.